src/redux/propertytree/propertyTreeSlice.ts

```diff
@@ -0,0 +1,194 @@
+      if (state.properties[uri]) {
+        state.properties[uri].value = value;
+      }
+      return state;
+    },
+    updatePropertyValue: (
+      state,
+      action: PayloadAction<{ uri: string; value: PropertyValue }>
+    ) => {
+      const { uri, value } = action.payload;
+
+      // TODO: check this return vs the old code
+      if (state.properties[uri]) {
+        state.properties[uri].value = value;
+      }
+      return state;
+
+      // const newPropertyState = { ...state.properties[action.payload.uri] };
+      // newPropertyState.value = action.payload.value;
+      // return {
+      //   ...state,
+      //   [action.payload.uri]: newPropertyState
+      // };
+    }
+  }
+});
+
+// Action creators are generated for each case reducer function, replaces the `Actions/index.js`
+export const {
+  addProperties,
+  removeProperties,
+  clearPropertyTree,
+  setPropertyValue,
+  updatePropertyValue
+} = properties.actions;
+
+// actions:
+// addPropertyOwners
+// addProperties
+// removePropertyOwners
+// removeProperties
+// updatePropertyValue
+// setPropertyValue
+// clearPropertyTree
+
+export interface PropertyOwnersState {
+  isInitialized: boolean;
+  propertyOwners: PropertyOwners;
+}
+
+const initialStatePropertyOwners: PropertyOwnersState = {
+  isInitialized: false,
+  propertyOwners: {}
+};
+
+export const propertyOwners = createSlice({
+  name: 'propertyOwners',
+  initialState: initialStatePropertyOwners,
+  reducers: {
+    addPropertyOwners: (
+      state,
+      action: PayloadAction<{ propertyOwners: PropertyOwner[] }>
+    ) => {
+      const inputOwners = action.payload.propertyOwners;
+      // const newState = { ...state };
+      inputOwners.forEach((owner) => {
+        state.propertyOwners[owner.uri] = {
+          identifier: owner.identifier,
+          name: owner.name,
+          properties: owner.properties,
+          subowners: owner.subowners,
+          tags: owner.tags || [],
+          uri: owner.uri, // TODO: Remove this
+          description: owner.description
+        };
+
+        // Ensure the parents of the uri have the links to the new entry
+        // Get parent uri
+        const periodPos = owner.uri.lastIndexOf('.');
+        const parentUri = owner.uri.substring(0, periodPos);
+
+        // If that parent exists and the link doesn't exist, add the link to the parent
+        const parentExists = parentUri && state.propertyOwners[parentUri];
+        if (
+          parentExists &&
+          !state.propertyOwners[parentUri]!.subowners.includes(owner.uri)
+        ) {
+          state.propertyOwners[parentUri]!.subowners.push(owner.uri);
+        }
+      });
+      return state;
+    },
+    clearPropertyTree: () => {
+      return {
+        isInitialized: false,
+        propertyOwners: {}
+      };
+    },
+    removePropertyOwners: (state, action: PayloadAction<{ uris: string[] }>) => {
+      // const newState = { ...state };
```

Comment:
:)

Reply:
Here's a comment on this behavior in a video, for future self
https://youtu.be/5yEG6GhoJBs?t=980

More notes to self: 
In summary, `createSlice` allows us to write "mutating code", but there is a bunch of magic happening behind the scene to create copy of the state and really only edit the copy. 

My first intuition is that we want to write the code more explicitly, by creating a copy like Ylva had started doing, to avoid confusion and reduce the need to know about the "magic"... But yeah let's discuss